integration/teleterm_test.go

```diff
@@ -0,0 +1,256 @@
+
+	"github.com/gravitational/teleport/api/types"
+	dbhelpers "github.com/gravitational/teleport/integration/db"
+	"github.com/gravitational/teleport/integration/helpers"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/service"
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
+	"github.com/gravitational/teleport/lib/teleterm/apiserver/handler"
+	"github.com/gravitational/teleport/lib/teleterm/clusters"
+	"github.com/gravitational/teleport/lib/teleterm/daemon"
+)
+
+func TestTeletermRPCService(t *testing.T) {
+	pack := dbhelpers.SetupDatabaseTest(t,
+		dbhelpers.WithListenerSetupDatabaseTest(helpers.SingleProxyPortSetup),
+		dbhelpers.WithLeafConfig(func(config *service.Config) {
+			config.Auth.NetworkingConfig.SetProxyListenerMode(types.ProxyListenerMode_Multiplex)
+		}),
+		dbhelpers.WithRootConfig(func(config *service.Config) {
+			config.Auth.NetworkingConfig.SetProxyListenerMode(types.ProxyListenerMode_Multiplex)
+		}),
+	)
+	pack.WaitForLeaf(t)
+
+	creds, err := helpers.GenerateUserCreds(helpers.UserCredsRequest{
+		Process:  pack.Root.Cluster.Process,
+		Username: pack.Root.User.GetName(),
+	})
+	require.NoError(t, err)
+
+	t.Run("adding root cluster", func(t *testing.T) {
+		t.Parallel()
+
+		testAddingRootCluster(t, pack, creds)
+	})
+
+	t.Run("list root clusters returning logged in user", func(t *testing.T) {
+		t.Parallel()
+
+		testIfListRootClustersReturnsLoggedInUser(t, pack, creds)
+	})
+	t.Run("get cluster returns properties from auth server", func(t *testing.T) {
+		t.Parallel()
+
+		testIfGetClusterReturnsPropertiesFromAuthServer(t, pack)
+	})
+}
+
+func testAddingRootCluster(t *testing.T, pack *dbhelpers.DatabasePack, creds *helpers.UserCreds) {
+	storage, err := clusters.NewStorage(clusters.Config{
+		Dir:                t.TempDir(),
+		InsecureSkipVerify: true,
+	})
+	require.NoError(t, err)
+
+	daemonService, err := daemon.New(daemon.Config{
+		Storage: storage,
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		daemonService.Stop()
+	})
+
+	addedCluster, err := daemonService.AddCluster(context.Background(), pack.Root.Cluster.Web)
+	require.NoError(t, err)
+
+	clusters, err := daemonService.ListRootClusters(context.Background())
+	require.NoError(t, err)
+
+	clusterURIs := make([]uri.ResourceURI, 0, len(clusters))
+	for _, cluster := range clusters {
+		clusterURIs = append(clusterURIs, cluster.URI)
+	}
+	require.ElementsMatch(t, clusterURIs, []uri.ResourceURI{addedCluster.URI})
+}
+
+func testIfListRootClustersReturnsLoggedInUser(t *testing.T, pack *dbhelpers.DatabasePack, creds *helpers.UserCreds) {
+	tc, err := simulateLogin(pack.Root.User.GetName(), pack, creds)
+	require.NoError(t, err)
+	// The profile on disk created by NewClientWithCreds doesn't have WebProxyAddr set.
+	tc.WebProxyAddr = pack.Root.Cluster.Web
+	tc.SaveProfile(false /* makeCurrent */)
```

Comment:
Oh nice, done

Reply:
I forgot we're inside the integration package so this `mustLogin` package is available to a lot of other tests. But it seems that no one thought about defining `mustLogin` so far anyway. ;f We could always rename it to something like `mustLoginClient` in the future so that the name is more specific.